File: cache-server/src/main/java/io/pivotal/demo/domain/WordCountWindowed.java
package io.pivotal.demo.domain;

import java.text.SimpleDateFormat;
import java.util.Date;

import org.springframework.data.annotation.Id;
import org.springframework.data.gemfire.mapping.annotation.Region;

@Region(name="Words_Count_Windowed")
public class WordCountWindowed {

	public WordCountWindowedCompsiteKey getWordCountCompsiteKey() {
		return wordCountWindowedCompsiteKey;
	}

	public void setWordCountCompsiteKey(WordCountWindowedCompsiteKey wordCountCompsiteKey) {
		this.wordCountWindowedCompsiteKey = wordCountCompsiteKey;
	}

	@Id
	WordCountWindowedCompsiteKey wordCountWindowedCompsiteKey;

	private long wordCountValue;

	private Date end;
	
	private Long endTimeInLong;
	
	private Long startTimeInLong;
	
	private String word;

	@Override
	public String toString() {
		SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
		final StringBuffer sb = new StringBuffer("WordCountWindowed{");
		sb.append("word='").append(wordCountWindowedCompsiteKey.getWord()).append('\'');
		sb.append(", count=").append(wordCountValue);
		sb.append(", start=").append(formatter.format(wordCountWindowedCompsiteKey.getStart()));
		sb.append(", end=").append(formatter.format(end));
		sb.append('}');
		return sb.toString();
	}

	public WordCountWindowed() {

	}

	public WordCountWindowed(WordCountWindowedCompsiteKey wordCountWindowedCompsiteKey, long count, 
			Date end, Long startTimeInLong, Long endTimeInLong) {
		this.wordCountWindowedCompsiteKey = wordCountWindowedCompsiteKey;
		this.wordCountValue = count;
		this.end = end;
		this.endTimeInLong = endTimeInLong;
		this.startTimeInLong = startTimeInLong;
		this.word = this.wordCountWindowedCompsiteKey.getWord();
	}
	
	public long getWordCountValue() {
		return wordCountValue;
	}

	public void setWordCountValue(long wordCountValue) {
		this.wordCountValue = wordCountValue;
	}

	public String getWord() {
		return word;
	}

	public void setWord(String word) {
		this.word = word;
	}

	public Date getEnd() {
		return end;
	}

	public void setEnd(Date end) {
		this.end = end;
	}
	
	public Long getEndTimeInLong() {
		return endTimeInLong;
	}

	public void setEndTimeInLong(Long endTimeInLong) {
		this.endTimeInLong = endTimeInLong;
	}

	public Long getStartTimeInLong() {
		return startTimeInLong;
	}

	public void setStartTimeInLong(Long startTimeInLong) {
		this.startTimeInLong = startTimeInLong;
	}
}

File: web-ui/src/main/kotlin/io/pivotal/demo/domain/WordCountWindowed.kt
package io.pivotal.demo.domain

import org.springframework.data.annotation.Id
import org.springframework.data.gemfire.mapping.annotation.Region

import java.text.SimpleDateFormat
import java.util.Date

@Region(name = "Words_Count_Windowed")
class WordCountWindowed {

    @Id
    var wordCountCompsiteKey: WordCountWindowedCompsiteKey? = null

    var wordCountValue: Long = 0

    var end: Date? = null

    var endTimeInLong: Long? = null

    var startTimeInLong: Long? = null

    var word: String? = null

    override fun toString(): String {
        val formatter = SimpleDateFormat("MM/dd/yyyy HH:mm:ss")
        val sb = StringBuffer("WordCountWindowed{")
        sb.append("word='").append(wordCountCompsiteKey?.word).append('\'')
        sb.append(", count=").append(wordCountValue)
        sb.append(", start=").append(formatter.format(wordCountCompsiteKey?.start))
        sb.append(", end=").append(formatter.format(end))
        sb.append('}')
        return sb.toString()
    }

    constructor(wordCountWindowedCompsiteKey: WordCountWindowedCompsiteKey?, count: Long,
                end: Date, startTimeInLong: Long?, endTimeInLong: Long?) {
        this.wordCountCompsiteKey = wordCountWindowedCompsiteKey
        this.wordCountValue = count
        this.end = end
        this.endTimeInLong = endTimeInLong
        this.startTimeInLong = startTimeInLong
        this.word = this.wordCountCompsiteKey?.word
    }

   constructor(){

   }
}

